src/pool/srv_target.c

```diff
@@ -237,21 +244,28 @@ pool_alloc_ref(void *key, unsigned int ksize, void *varg,
 		goto err_iv_lock;
 	}
 
-	if (arg->pca_need_group) {
-		char id[DAOS_UUID_STR_SIZE];
+	uuid_unparse_lower(key, group_id);
+	rc = crt_group_secondary_create(group_id, NULL /* primary_grp */,
+					NULL /* ranks */, &pool->sp_group);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create pool group: %d\n",
+			DP_UUID(key), rc);
+		goto err_collective;
+	}
 
-		uuid_unparse_lower(key, id);
-		pool->sp_group = crt_group_lookup(id);
-		if (pool->sp_group == NULL) {
-			D_ERROR(DF_UUID": pool group not found\n",
-				DP_UUID(key));
-			D_GOTO(err_collective, rc = -DER_NONEXIST);
-		}
+	rc = ds_iv_ns_create(info->dmi_ctx, pool->sp_uuid, pool->sp_group,
+			     &iv_ns_id, &pool->sp_iv_ns);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create pool IV NS: %d\n",
+			DP_UUID(key), rc);
+		goto err_group;
```

Comment:
Do you consider to move these iv_ns_create/pool_create/collective outside of pool_alloc_ref(), which is really confusing.

Reply:
ds_pool was designed to host the VOS pool handles and the group pointer... Could we have a chat?